### DISCUSSION TITLE
Complejidad y Hashes Parte 1

### DISCUSSION THREAD

**Discussion by Cristian1754** [IIC2133-PUC/2024-1#169] [Tarea 2] [answered]
Hola, mi duda es la siguiente, he notado que para cumplir con la complejidad solicitada la función hash no puede tener colisiones. Esto es debido a que si tengo dos strings con el mismo hash, no puedo asumir que son el mismo y se debe comparar caracter por caracter (si hubiera otra forma, ese seria otro hash). Tomando esto en consideración, en el peor caso todos los substrings tienen el mismo hash y la busqueda me llevaria O(MxN) por cada consulta, es decir O(NxMxQ) en total. ¿Es este razonamiento correcto o el caso que me planteo es muy raro como para considerarlo?

**Comment by sebacarrascop** [comment, marked as answer]
hola, no hay problema con que tengas colisiones!, ya que la búsqueda entre los elementos "colisionados" tiene una complejidad menor en magnitudes, entonces la complejidad del hash no se ve afectada por esa búsqueda.
claramente eso es en lo teórico, si tienes demasiadas colisiones eso puede afectar el rendimiento en la practica, para que tengas ojo.

**Reply by Cristian1754** [reply]
Claro, entonces no hay problema en comparar manualmente cuando hay colisiones (para esto el hash debe ser suficientemente bueno como para que no hayan tantas colisiones)

**Reply by sebacarrascop** [reply]
exacto!, saludos
